tests/connection/test_fsm.py

```diff
@@ -0,0 +1,139 @@
+import pytest
+
+from supa.connection.fsm import LifecycleStateMachine, ProvisionStateMachine, ReservationStateMachine
+from supa.db.model import Reservation
+
+
+def test_reservation_state_machine() -> None:  # noqa: D103
+    reservation = Reservation()
+    rsm = ReservationStateMachine(reservation, state_field="reservation_state")
+    #
+    # reserve_request -> reserve_failed -> reserve_abort_request -> reserve_abort_confirmed
+    #
+    assert reservation.reservation_state == ReservationStateMachine.ReserveStart.value
+    rsm.reserve_request()
+    assert reservation.reservation_state == ReservationStateMachine.ReserveChecking.value
+    rsm.reserve_failed()
```

Comment:
I see some inconsistencies in naming:
Provisioning -> Provisioned

But for reserving stuff another naming strategy seems to be used. Not sure if this is intentional.

Reply:
I can imagine that this seems inconsistent, but the naming comes from the [NSI Connection Service v2.1](https://www.ogf.org/documents/GFD.237.pdf) standards document paragraph 5.3. A distinction is made between initial, transitional and stable states. The states ending in ...ing are transitional states.